pos.lua

```diff
@@ -47,20 +47,28 @@ minetest.register_chatcommand("area_pos1", {
 	privs = {},
 	func = function(name, param)
 		local pos
-		local found, _, x, y, z = param:find(
-				"^(-?%d+)[, ](-?%d+)[, ](-?%d+)$")
+		local player = minetest.get_player_by_name(name)
+		if player then
+			pos = vector.round(player:get_pos())
+		end
+		local found, _, x_str, y_str, z_str = param:find(
+			"^(~?-?%d*)[, ](~?-?%d*)[, ](~?-?%d*)$")
 		if found then
-			pos = {x=tonumber(x), y=tonumber(y), z=tonumber(z)}
-		elseif param == "" then
-			local player = minetest.get_player_by_name(name)
-			if player then
-				pos = player:get_pos()
-			else
-				return false, S("Unable to get position.")
+			local x = pos and minetest.parse_relative_number(x_str, pos.x)
```

Comment:
This needs Minetest 5.7.0 or newer. Please add a static guard to provide a helpful error message to those who are using an older version - or make the code backwards-compatible.

Reply:
Fixed in ded79e9

Reply:
Fixed in a9b2c7e